### ISSUE TITLE
Hi from Lava Network :) typescript type definitions are failing.

### ISSUE THREAD

**Issue by ranlavanet** [cosmology-tech/telescope#450]
Hi, we are trying to create lavajs on our lava protobufs, previous versions seemed to work suddenly all types fail 

<img width="824" alt="image" src="https://github.com/cosmology-tech/telescope/assets/106548467/5d53e50f-3317-471b-b581-37a2eb68e72f">

steps we run:

telescope install -> cosmos / tendermint / gogo / google / cosmos_proto / amino 
copy our lava repo protobufs 

transpile (successfully) 

build -> fails to generate .d.ts files

we use this script: 

https://github.com/lavanet/lava/pull/721/files#diff-6271b884941c609cec6724a16d13df4c19f35406f91f25dfbcb88c1a10539001

basically transpiling and building. nothing else.

**Comment by Zetazzz** [comment]
Hi, thanks for feedback! I'll look into this asap, possibly a config issue.

**Comment by Zetazzz** [comment]
This config possibly solve the issue, please give it a try. If it's not I'll look into more. Thank you so much!

aminoEncoding: {
      enabled: true,
      exceptions: AMINO_MAP,
      useRecursiveV2encoding: true
    },

**Comment by ranlavanet** [comment]
tried adding this: 
<img width="328" alt="image" src="https://github.com/cosmology-tech/telescope/assets/106548467/2111a6e3-2ba9-4a3a-ad56-1ddbc60661cd">

but it didn't resolve the issue. just an FYI telescope did work before the new repo change for us

still getting issues also for cosmos and tendermint repositories alone:

```src/codegen/tendermint/version/types.ts:96:25 - error TS2304: Cannot find name 'AppProtoMsg'.

96   fromProtoMsg(message: AppProtoMsg): App {
                           ~~~~~~~~~~~

src/codegen/tendermint/version/types.ts:102:29 - error TS2304: Cannot find name 'AppProtoMsg'.

102   toProtoMsg(message: App): AppProtoMsg {
                                ~~~~~~~~~~~

src/codegen/tendermint/version/types.ts:152:21 - error TS2304: Cannot find name 'ConsensusAmino'.

152   fromAmino(object: ConsensusAmino): Consensus {
                        ~~~~~~~~~~~~~~

src/codegen/tendermint/version/types.ts:158:32 - error TS2304: Cannot find name 'ConsensusAmino'.

158   toAmino(message: Consensus): ConsensusAmino {
                                   ~~~~~~~~~~~~~~

src/codegen/tendermint/version/types.ts:164:24 - error TS2304: Cannot find name 'ConsensusAminoMsg'.

164   fromAminoMsg(object: ConsensusAminoMsg): Consensus {
                           ~~~~~~~~~~~~~~~~~

src/codegen/tendermint/version/types.ts:167:25 - error TS2304: Cannot find name 'ConsensusProtoMsg'.

167   fromProtoMsg(message: ConsensusProtoMsg): Consensus {
                            ~~~~~~~~~~~~~~~~~

src/codegen/tendermint/version/types.ts:173:35 - error TS2304: Cannot find name 'ConsensusProtoMsg'.

173   toProtoMsg(message: Consensus): ConsensusProtoMsg {
                                      ~~~~~~~~~~~~~~~~~

```

**Comment by ranlavanet** [comment]
Update: 

instead of using the telescope binary

I ran it with the codegen.js file. 



now getting much less issues 

```
$ /home/user/go/src/lava/ecosystem/lavajs/node_modules/.bin/tsc -p tsconfig.json --outDir mjs --module es2022 --declaration false
src/codegen/cosmos/staking/v1beta1/staking.ts:1293:7 - error TS2322: Type 'string' is not assignable to type 'Any'.

1293       consensusPubkey: encodeBech32Pubkey({
           ~~~~~~~~~~~~~~~

  src/codegen/cosmos/staking/v1beta1/staking.ts:221:3
    221   consensusPubkey: Any;
          ~~~~~~~~~~~~~~~
    The expected type comes from property 'consensusPubkey' which is declared here on type 'Validator'

src/codegen/cosmos/staking/v1beta1/staking.ts:1313:44 - error TS2345: Argument of type 'Any' is not assignable to parameter of type 'string'.

1313       value: fromBase64(decodeBech32Pubkey(message.consensusPubkey).value)
                                                ~~~~~~~~~~~~~~~~~~~~~~~

src/codegen/cosmos/staking/v1beta1/tx.ts:367:7 - error TS2322: Type 'string' is not assignable to type 'Any'.

367       pubkey: encodeBech32Pubkey({
          ~~~~~~

  src/codegen/cosmos/staking/v1beta1/tx.ts:16:3
    16   pubkey: Any;
         ~~~~~~
    The expected type comes from property 'pubkey' which is declared here on type 'MsgCreateValidator'

src/codegen/cosmos/staking/v1beta1/tx.ts:383:44 - error TS2345: Argument of type 'Any' is not assignable to parameter of type 'string'.

383       value: fromBase64(decodeBech32Pubkey(message.pubkey).value)
                                               ~~~~~~~~~~~~~~

src/codegen/cosmos/tx/signing/v1beta1/signing.ts:319:11 - error TS2304: Cannot find name 'Data'.

319     data: Data.fromPartial({}),
```

**Comment by Zetazzz** [comment]
> Update:
> 
> instead of using the telescope binary
> 
> I ran it with the codegen.js file.
> 
> now getting much less issues
> 
> ```
> $ /home/user/go/src/lava/ecosystem/lavajs/node_modules/.bin/tsc -p tsconfig.json --outDir mjs --module es2022 --declaration false
> src/codegen/cosmos/staking/v1beta1/staking.ts:1293:7 - error TS2322: Type 'string' is not assignable to type 'Any'.
> 
> 1293       consensusPubkey: encodeBech32Pubkey({
>            ~~~~~~~~~~~~~~~
> 
>   src/codegen/cosmos/staking/v1beta1/staking.ts:221:3
>     221   consensusPubkey: Any;
>           ~~~~~~~~~~~~~~~
>     The expected type comes from property 'consensusPubkey' which is declared here on type 'Validator'
> 
> src/codegen/cosmos/staking/v1beta1/staking.ts:1313:44 - error TS2345: Argument of type 'Any' is not assignable to parameter of type 'string'.
> 
> 1313       value: fromBase64(decodeBech32Pubkey(message.consensusPubkey).value)
>                                                 ~~~~~~~~~~~~~~~~~~~~~~~
> 
> src/codegen/cosmos/staking/v1beta1/tx.ts:367:7 - error TS2322: Type 'string' is not assignable to type 'Any'.
> 
> 367       pubkey: encodeBech32Pubkey({
>           ~~~~~~
> 
>   src/codegen/cosmos/staking/v1beta1/tx.ts:16:3
>     16   pubkey: Any;
>          ~~~~~~
>     The expected type comes from property 'pubkey' which is declared here on type 'MsgCreateValidator'
> 
> src/codegen/cosmos/staking/v1beta1/tx.ts:383:44 - error TS2345: Argument of type 'Any' is not assignable to parameter of type 'string'.
> 
> 383       value: fromBase64(decodeBech32Pubkey(message.pubkey).value)
>                                                ~~~~~~~~~~~~~~
> 
> src/codegen/cosmos/tx/signing/v1beta1/signing.ts:319:11 - error TS2304: Cannot find name 'Data'.
> 
> 319     data: Data.fromPartial({}),
> ```

de(en)codeBech32Pubkey related errors have been fixed in the version:
telescope@0.105.0

**Comment by ranlavanet** [comment]
thanks works with the latest codegen.js release :) closing issue
